### DISCUSSION TITLE
Has anyone been able to successfully archive an Instagram post?

### DISCUSSION THREAD

**Discussion by ghost** [ArchiveBox/ArchiveBox#1471] [Q&A]
Here's the configurations I've tried:

- TIMEOUT=30 (to load all the dynamic elements of the page)
- CHROME_USER_DATA_DIR=
- COOKIES_FILE=
- WGET_USER_AGENT="Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36"

And here are some screenshots of what I get:
![instagram_1](https://github.com/user-attachments/assets/a4d00e28-7ee2-49e4-95ae-5dd2abec76c0)
![instagram_2](https://github.com/user-attachments/assets/2defac49-32e3-437e-9846-2c829a41896b)

**Comment by pirate** [comment]
Can you try saving instagram with https://github.com/gildas-lormeau/single-file-cli directly, outside of ArchiveBox?

**Reply by ghost** [reply]
Yep, that works. But how would you save a WARC of the web page?

**Reply by pirate** [reply]
If you need a WARC for instagram you should use ArchiveWeb.page / browsertrix-crawler instead. ArchiveBox cannot capture a working WARC of Instagram yet because we use wget to generate the WARCs currently, and it's not advanced enough to handle big social media sites. Singlefile is the only method that typically works in ArchiveBox for FB/Insta/TikTok/etc, but Singlefile's output is just a `singlefile.html` not a warc.

I have an archiving engine I wrote for a paying ArchiveBox client with a fairly advanced puppeteer setup with lots of stealth / bot detection avoidance stuff, but I haven't merged it back into ArchiveBox core yet as I need to finish the 0.8.0 release first. There will be progress improving the state of social media WARCs at some point, but I can't promise specific dates yet.

**Reply by ghost** [reply]
Oh, ok. Yeah, I can see how big social media sites can be a pain for archiving. But thanks for the ArchiveWeb.page recommendation.
